themes/openy_themes/openy_rose/templates/node/node--branch--teaser.html.twig

```diff
@@ -90,9 +90,11 @@
       {% set title_url = url %}
     {% endif %}
 
-    <h2{{ title_attributes }}>
-      <a href="{{ title_url }}" rel="bookmark">{{ label }}</a>
-    </h2>
+    <div class="associated-branch-title">
```

Comment:
This will remove headings for all branch teasers on all pages.
Please create separate template for this view mode.